flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-paimon/src/main/java/org/apache/flink/cdc/connectors/paimon/sink/PaimonMetadataApplier.java

```diff
@@ -129,25 +130,127 @@ private void applyCreateTable(CreateTableEvent event)
+                case BEFORE:
+                    SchemaChange schemaChangeWithBeforePosition =
+                            applyAddColumnWithBeforePosition(
+                                    event.tableId().getSchemaName(),
+                                    event.tableId().getTableName(),
+                                    columnWithPosition);
+                    tableChangeList.add(schemaChangeWithBeforePosition);
+                    break;
+                case AFTER:
+                    if (columnWithPosition.getExistedColumnName() == null) {
+                        throw new IllegalArgumentException(
+                                "Existing column name must be provided for AFTER position");
+                    }
+                    tableChange =
+                            SchemaChange.addColumn(
+                                    columnWithPosition.getAddColumn().getName(),
+                                    LogicalTypeConversion.toDataType(
+                                            DataTypeUtils.toFlinkDataType(
+                                                            columnWithPosition
+                                                                    .getAddColumn()
+                                                                    .getType())
+                                                    .getLogicalType()),
+                                    columnWithPosition.getAddColumn().getComment(),
+                                    SchemaChange.Move.after(
+                                            columnWithPosition.getAddColumn().getName(),
+                                            columnWithPosition.getExistedColumnName()));
+                    tableChangeList.add(tableChange);
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                            "Unknown column position: " + columnWithPosition.getPosition());
+            }
+        }
+        return tableChangeList;
+    }
+
+    private SchemaChange applyAddColumnWithLastPosition(
+            String schemaName,
+            String tableName,
+            AddColumnEvent.ColumnWithPosition columnWithPosition)
+            throws Catalog.TableNotExistException {
+        Table table = catalog.getTable(new Identifier(schemaName, tableName));
+        List<String> columnNames = table.rowType().getFieldNames();
+        String lastColumnName = columnNames.get(columnNames.size() - 1);
+        return SchemaChange.addColumn(
+                columnWithPosition.getAddColumn().getName(),
+                LogicalTypeConversion.toDataType(
+                        DataTypeUtils.toFlinkDataType(columnWithPosition.getAddColumn().getType())
+                                .getLogicalType()),
+                columnWithPosition.getAddColumn().getComment(),
+                SchemaChange.Move.after(
+                        columnWithPosition.getAddColumn().getName(), lastColumnName));
```

Comment:
`SchemaChange.Move.after(columnWithPosition.getAddColumn().getName(), lastColumnName)` can be removed as it's the default behavior of [SchemaManager](https://github.com/apache/paimon/blob/master/paimon-core/src/main/java/org/apache/paimon/schema/SchemaManager.java).